core/valor_core/text_generation.py

```diff
@@ -0,0 +1,634 @@
+                            "context_list": prediction_context_list,
+                        },
+                    )
+                ]
+
+            if is_SummaryCoherence_enabled:
+                score = client.summary_coherence(
+                    text=datum_text,
+                    summary=prediction_text,
+                )
+                output += [
+                    metrics.SummaryCoherenceMetric(
+                        value=score,
+                        parameters={
+                            "datum_uid": datum_uid,
+                            "prediction": prediction_text,
+                        },
+                    )
+                ]
+
+            if is_Toxicity_enabled:
+                score = client.toxicity(text=prediction_text)
+                output += [
+                    metrics.ToxicityMetric(
+                        value=score,
+                        parameters={
+                            "datum_uid": datum_uid,
+                            "prediction": prediction_text,
+                        },
+                    )
+                ]
+
+    return output
+
+
+def evaluate_text_generation(
+    predictions: list[schemas.Prediction],
+    metrics_to_return: list[MetricType],
+    groundtruths: list[schemas.GroundTruth] = [],
+    llm_api_params: dict[str, str | dict] | None = None,
+    metric_params: dict[str, dict] = {},
+) -> schemas.Evaluation:
+    """
+    Validates the parameters and formats the predictions and ground truths, then computes the text generation metrics.
+
+    Parameters
+    ----------
+    predictions : list[schemas.Prediction]
+        A list of predictions.
+    metrics_to_return : list[MetricType]
+        The list of metrics to compute, store, and return to the user. There is no default value, so the user must specify the metrics they want to compute.
+    groundtruths : list[schemas.GroundTruth], optional
+        A list of ground truths. Ground truths are not required for all text generation metrics.
+    llm_api_params : dict[str, str | dict], optional
+        A dictionary of parameters for the LLM API.
+    metric_params : dict, optional
+        A dictionary of optional parameters to pass in to specific metrics.
+
+    Returns
+    ----------
+    schemas.Evaluation
+        An evaluation object containing the computed metrics and metadata.
+    """
+    start_time = time.time()
+
+    utilities.validate_metrics_to_return(
+        metrics_to_return=metrics_to_return,
+        task_type=enums.TaskType.TEXT_GENERATION,
+    )
+    utilities.validate_metric_parameters(
+        metrics_to_return=metrics_to_return,
+        metric_params=metric_params,
+    )
+
+    unique_datum_counts = len(set([p.datum.uid for p in predictions]))
+
+    # Generate a list of data tuples, where each prediction is matched with its corresponding datum and the associated groundtruths.
+    # TODO consider changing to dataframe
+    data = []
```

Comment:
I was considering using a pandas dataframe like the other task types use. However I don't think this code needs a dataframe for any reason (I'm not doing any computations that would benefit from having a dataframe), so I don't think this change would be worth it for now.